main.go

```diff
@@ -100,6 +100,21 @@ func main() {
 	}
 
 	osClient := osclientset.NewForConfigOrDie(rest.AddUserAgent(config, controllers.ComponentName))
+	// Check the Platform Type to determine the state of the CO
+	enabled, err := controllers.IsEnabled(osClient)
+	if err != nil {
+		setupLog.Error(err, "could not determine whether to run")
+		os.Exit(1)
+	}
+	if !enabled {
+		//Set ClusterOperator status to disabled=true, available=true
+		err = controllers.SetCOInDisabledState(osClient, releaseVersion)
```

Comment:
I think you should move the content of SetCOInDisabledState() into updateCOStatus() - if there is anything missing
```suggestion
		err = controllers.updateCOStatus(...)
```

Reply:
also, you could do this within SetupWithManager so that you have the reconcile object